io/fv3atm_sfc_io.F90

```diff
@@ -255,7 +255,8 @@ subroutine Sfc_io_register_axes(sfc, Model, Sfc_restart, reading, warm_start)
       if (Model%lsm == Model%lsm_noah .or. Model%lsm == Model%lsm_noahmp) then
         call register_axis(Sfc_restart, 'zaxis_2', dimension_length=Model%lsoil)
       else if(Model%lsm == Model%lsm_ruc .and. reading) then
-        ! Possible bug. This is only defined on read, not write.
+        ! The RUC defines zaxis_1 for reading, so it can read a restart from a different LSM.
+        ! The lsm_ruc never writes zaxis_2.
```

Comment:
@SamuelTrahanNOAA Could you please explain in the comment what is zaxis_1 and zaxis_2? Thank you!

Reply:
I'm working my way through the code, adding comments, right now. I'll be sure to add that one.

Reply:
There is a much longer paragraph explaining the situation. How does it look now?

```f90
        call register_axis(Sfc_restart, 'zaxis_2', dimension_length=Model%lsoil_lsm)
        ! The RUC only ever writes zaxis_1, which is combined soil-ice, kice, which is 9.
        ! Other LSMs read and write zaxis_2, which is lsoil for them, and that's always 4.
        ! Defining zaxis_2 here lets RUC LSM read from a different soil vertical coordinate
        ! (lsoil_lsm). In practice, this probably won't work well since levels will be
        ! filled with zeroes. We retain this capability for historical reasons.
        ! Just make sure you only restart RUC LSM off of RUC LSM, and always have
        ! kice = lsoil = lsoil_lsm = 9 and everything will be fine.
```

Reply:
@SamuelTrahanNOAA Thank you very much for adding detailed description here. It looks good to me, although I think the sentence in bold could be confusing. 
        ! The RUC only ever writes zaxis_1, which is combined soil-ice, kice, which is 9.
        ! Other LSMs read and write zaxis_2, which is lsoil for them, and that's always 4.
        ! Defining zaxis_2 here lets RUC LSM read from a different soil vertical coordinate
        ! (lsoil_lsm). **In practice, this probably won't work well since levels will be
        ! filled with zeroes.** We retain this capability for historical reasons.
        ! Just make sure you only restart RUC LSM off of RUC LSM, and always have
        ! kice = lsoil = lsoil_lsm = 9 and everything will be fine.

I propose the following:        
        ! The RUC only ever writes zaxis_1, which is combined soil/ice vertical dimension, lsoil_lsm/kice, which is 9.
        ! Other LSMs read and write zaxis_2, which is lsoil for them, and that's always 4.
        ! Defining zaxis_2 here lets RUC LSM read from a different soil vertical coordinate
        ! (lsoil_lsm). It is needed for restart of RUC LSM from RUC LSM. This capability exists for historical reasons,
        ! because there are two sets of soil state variables: one set has lsoil=4 vertical layers (Noah LSM. NoahMP LSM), 
        ! and another set has lsoil_lsm=9 vertical levels (RUC LSM). Ideally there should be just one set of soil variables
        ! that could have different vertical dimension depending on the choice of LSM.
        ! For now: just make sure you only restart RUC LSM off of RUC LSM, and always have
        ! kice = lsoil = lsoil_lsm = 9 and everything will be fine.

Reply:
I have replaced my comment with yours and I word-wrapped it. How does it look?

Reply:
Looks great. Thank you very much, Sam.